src/test/shader.js

```diff
@@ -0,0 +1,640 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+/* eslint-disable no-var, max-statements */
+import 'babel-polyfill';
+import {document, window} from 'global';
+import {Buffer, createGLContext, Program, glGetDebugInfo} from 'luma.gl';
+
+const glslify = require('glslify');
+
+// Utilities functions that to be moved to a common place for future tests
+
+function glEnumToString(gl, value) {
+  // Optimization for the most common enum:
+  if (value === gl.NO_ERROR) {
+    return "NO_ERROR";
+  }
+  for (var p in gl) {
+    if (gl[p] == value) {
+      return p;
+    }
+  }
+  return "0x" + value.toString(16);
+};
+
+function addSpan(contents, div) {
+  if (div == undefined) {
+    var divs = document.body.getElementsByClassName("testInfo");
+    var lastDiv = divs[divs.length - 1];
+    div = lastDiv;
+  }
+
+  var span = document.createElement("span");
+  div.appendChild(span);
+  span.innerHTML = contents + '<br />';
+}
+
+function addDiv(contents) {
+    var testInfoDiv = document.createElement("div");
+    document.body.appendChild(testInfoDiv);
+    testInfoDiv.setAttribute("class", "testInfo");
+
+    return testInfoDiv;
+}
+
+function logToConsole(msg) {
+  if (window.console)
+    window.console.log(msg);
+}
+
+function escapeHTML(text) {
+  return text.replace(/&/g, "&amp;").replace(/</g, "&lt;");
+}
+
+function testPassed(msg) {
+  addSpan('<span><span class="pass" style="color:green">PASS</span> ' + escapeHTML(msg) + '</span>');
+  logToConsole('PASS ' + msg);
+}
+
+function testFailed(msg) {
+  addSpan('<span><span class="fail" style="color:red">FAIL</span> ' + escapeHTML(msg) + '</span>');
+  logToConsole('FAIL ' + msg);
+}
+
+function glErrorShouldBe(gl, glErrors, opt_msg) {
+  if (!glErrors.length) {
+    glErrors = [glErrors];
+  }
+  opt_msg = opt_msg || "";
+  var err = gl.getError();
+  var ndx = glErrors.indexOf(err);
+  var errStrs = [];
+  for (var ii = 0; ii < glErrors.length; ++ii) {
+    errStrs.push(glEnumToString(gl, glErrors[ii]));
+  }
+  var expected = errStrs.join(" or ");
+  if (ndx < 0) {
+    var msg = "getError expected" + ((glErrors.length > 1) ? " one of: " : ": ");
+    testFailed(msg + expected +  ". Was " + glEnumToString(gl, err) + " : " + opt_msg);
+  } else {
+    //var msg = "getError was " + ((glErrors.length > 1) ? "one of: " : "expected value: ");
+    //testPassed(msg + expected + " : " + opt_msg);
+  }
+};
+
+// Special utility functions for df64 tests
```

Comment:
These shouldn't be in here, right? They should be in a general utility file for 64 bit floats.